File: 2000/01/main.c
#include "../../ext/c/lib.h"

char *part1_result(char *data[])
{
  return data[0];
}

char *part2_result(char *data[])
{
  return data[1];
}

int main(int argc, char *argv[])
{
  char *data[2];
  data[0] = "Hello";
  data[1] = "World!";

  executor(data, part1_result, part2_result, argc, argv);
}

File: 2000/01/main.rb
require 'aoc_executor'

part1_proc = Proc.new { |input| input[0] }
part2_proc = Proc.new { |input| input[1] }
executor = AocExecutor.new(%w[Hello World!], part1_proc, part2_proc)
executor.run(ARGV)
